clusterloader2/pkg/prometheus/prometheus.go

```diff
@@ -118,18 +133,60 @@ func NewPrometheusController(clusterLoaderConfig *config.ClusterLoaderConfig) (p
 		clusterLoaderConfig.PrometheusConfig.ScrapeKubeProxy = mapping["PROMETHEUS_SCRAPE_KUBE_PROXY"].(bool)
 	}
 	mapping["PROMETHEUS_SCRAPE_KUBELETS"] = clusterLoaderConfig.PrometheusConfig.ScrapeKubelets
+	// Add PROVIDER to template mapping for conditions control in prometheus-prometheus.yaml
+	if _, exists := mapping["PROVIDER"]; !exists {
+		mapping["PROVIDER"] = clusterLoaderConfig.ClusterConfig.Provider
+	}
+	// If enabled prometheus scrape windows node, need to fetch windows node name as target
+	if enableScrapingWindowsNode(mapping, clusterLoaderConfig) {
+		windowsNode, err = getGceWindowsNodeNameFromKubernetesService(pc.framework.GetClientSets().GetClient())
+		if err != nil {
+			return nil, err
+		}
+		mapping["WINDOWS_NODE_NAME"] = windowsNode.nodeName
+	}
+
 	pc.templateMapping = mapping
 
 	return pc, nil
 }
 
+func enableScrapingWindowsNode(mapping map[string]interface{}, clusterLoaderConfig *config.ClusterLoaderConfig) bool {
+	if windowsNodeTest, exists := mapping["WINDOWS_NODE_TEST"]; exists && windowsNodeTest.(bool) && clusterLoaderConfig.ClusterConfig.Provider == "gce" {
+		return true
+	}
+	return false
+}
+
+func installWmiExporter(k8sClient kubernetes.Interface) error {
+	InstallWmiCmdTemplate := `Start-Job -Name InstallWmiExporter -ScriptBlock {invoke-webrequest -uri %s -outfile C:\wmi_exporter.msi; Start-Process msiexec.exe -Wait -ArgumentList '/I C:\wmi_exporter.msi ENABLED_COLLECTORS=%s LISTEN_PORT=5000 /quiet'}; Wait-Job -Name InstallWmiExporter`
+	InstallWmiExporterCmd := fmt.Sprintf(InstallWmiCmdTemplate, wmiExporterURL, wmiExporterCollectors)
+	powershellCmd := fmt.Sprintf(`powershell.exe -Command "%s"`, InstallWmiExporterCmd)
+	if windowsNode == nil {
```

Comment:
It's weird that windowsNode is a global variable. Is there a way to pass it as an arg here?

Reply:
As I separated the steps of 1) fetching windowsNode object to fill the windows scrape configs yaml, 2) install the wmi exporter from windowsNode object, so need a global object to share windows node info. Now combined them together, doesn't need it anymore.